code/modules/power/supermatter/supermatter.dm

```diff
@@ -413,21 +419,34 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		bzcomp += clamp(max(removed.gases[/datum/gas/bz][MOLES]/combined_gas, 0) - bzcomp, -1, gas_change_rate)
 		n2ocomp += clamp(max(removed.gases[/datum/gas/nitrous_oxide][MOLES]/combined_gas, 0) - n2ocomp, -1, gas_change_rate)
 		n2comp += clamp(max(removed.gases[/datum/gas/nitrogen][MOLES]/combined_gas, 0) - n2comp, -1, gas_change_rate)
+		h2ocomp += clamp(max(removed.gases[/datum/gas/water_vapor][MOLES]/combined_gas, 0) - h2ocomp, -1, gas_change_rate)
 
 		//We're concerned about pluoxium being too easy to abuse at low percents, so we make sure there's a substantial amount.
 		if(pluoxiumcomp >= 0.15)
 			pluoxiumbonus = 1    //makes pluoxium only work at 15%+
 		else
 			pluoxiumbonus = 0
 
+		if(h2ocomp >= 0.45 && h2ocomp <= 0.55) //heat protection from h2o only works when it's between 45 and 55%
+			h2obonus = 1
+		else
+			h2obonus = 0
+
+		if(h2ocomp >= 0.15 && h2ocomp <= 0.55) //the engine will stop producing power from 15% to 55% h2o (slow down the o2 and plasma generation too)
+			h2omalus = 0.1
+		else if (h2ocomp > 0.07 && h2ocomp< 0.15 || h2ocomp > 55) //above or below these 2 values the engine will start to freak out (may need test on the amount of freak out)
+			h2omalus = 2
+		else
+			h2omalus = 1
+
 		//No less then zero, and no greater then one, we use this to do explosions and heat to power transfer
-		gasmix_power_ratio = min(max(plasmacomp + o2comp + co2comp + tritiumcomp + bzcomp - pluoxiumcomp - n2comp, 0), 1)
+		gasmix_power_ratio = min(max(((plasmacomp + o2comp + co2comp + h2ocomp + tritiumcomp + bzcomp - pluoxiumcomp - n2comp) / h2omalus), 0), 1)
 		//Minimum value of 1.5, maximum value of 15
-		dynamic_heat_modifier = max((plasmacomp * PLASMA_HEAT_PENALTY) + (o2comp * OXYGEN_HEAT_PENALTY) + (co2comp * CO2_HEAT_PENALTY) + (tritiumcomp * TRITIUM_HEAT_PENALTY) + ((pluoxiumcomp * PLUOXIUM_HEAT_PENALTY) * pluoxiumbonus) + (n2comp * NITROGEN_HEAT_PENALTY) + (bzcomp * BZ_HEAT_PENALTY), 0.5)
+		dynamic_heat_modifier = max(((plasmacomp * PLASMA_HEAT_PENALTY) + (h2ocomp * H2O_HEAT_PENALTY) + (o2comp * OXYGEN_HEAT_PENALTY)) * h2omalus + (co2comp * CO2_HEAT_PENALTY) + (tritiumcomp * TRITIUM_HEAT_PENALTY) + ((pluoxiumcomp * PLUOXIUM_HEAT_PENALTY) * pluoxiumbonus) + (n2comp * NITROGEN_HEAT_PENALTY) + (bzcomp * BZ_HEAT_PENALTY), 0.5)
```

Comment:
I think I see what you're trying to do here, why not expand it to all positive values? Also could you update the comment or add one somewhere describing what is going on?

Reply:
Oh and update the max value in the comment to 21.5

Reply:
Actually it looks like 27.8 is the max. See dotplot. https://www.desmos.com/calculator/8ncpgbjyzd

Reply:
And the last comment that needs changing, could you update this line to 2500 * 27.8.
https://github.com/tgstation/tgstation/blob/fb169f8237a7cb7fc3218d065978516d1fa76da9/code/modules/power/supermatter/supermatter.dm#L505

Reply:
wOOOP I'm dumb. This should need it too.
https://github.com/tgstation/tgstation/blob/fb169f8237a7cb7fc3218d065978516d1fa76da9/code/modules/power/supermatter/supermatter.dm#L503

Reply:
updated, also i want to try and keep it only for o2, plasma and h2o since those will be the dangerous gases, but i could change it to include the other stuff too